OptimizelySDK.Tests/NotificationTests/NotificationCenterTests.cs

```diff
@@ -0,0 +1,251 @@
+
+            // Verify that callbacks added successfully.
+            Assert.AreEqual(3, NotificationCenter.NotificationsCount);
+
+            // Verify that only decision callbacks are removed.
+            NotificationCenter.ClearNotifications(NotificationTypeActivate);
+            Assert.AreEqual(1, NotificationCenter.NotificationsCount);
+            
+            // Verify that ClearNotifications does not break on calling twice for same type.
+            NotificationCenter.ClearNotifications(NotificationTypeActivate);
+            NotificationCenter.ClearNotifications(NotificationTypeActivate);
+
+            // Verify that ClearNotifications does not break after calling ClearAllNotifications.
+            NotificationCenter.ClearAllNotifications();
+            NotificationCenter.ClearNotifications(NotificationTypeTrack);
+        }
+
+        [Test]
+        public void TestClearAllNotifications()
+        {
+            // Add decision notifications.
+            NotificationCenter.AddNotification(NotificationTypeActivate, TestNotificationCallbacks.TestActivateCallback);
+            NotificationCenter.AddNotification(NotificationTypeActivate, TestNotificationCallbacks.TestAnotherActivateCallback);
+
+            // Add track notification.
+            NotificationCenter.AddNotification(NotificationTypeTrack, TestNotificationCallbacks.TestTrackCallback);
+
+            // Verify that callbacks added successfully.
+            Assert.AreEqual(3, NotificationCenter.NotificationsCount);
+
+            // Verify that ClearAllNotifications remove all the callbacks.
+            NotificationCenter.ClearAllNotifications();
+            Assert.AreEqual(0, NotificationCenter.NotificationsCount);
+
+            // Verify that ClearAllNotifications does not break on calling twice or after ClearNotifications.
+            NotificationCenter.ClearNotifications(NotificationTypeActivate);
+            NotificationCenter.ClearAllNotifications();
+            NotificationCenter.ClearAllNotifications();
+        }
+
+        [Test]
+        public void TestSendNotifications()
+        {
+            var config = ProjectConfig.Create(TestData.Datafile, LoggerMock.Object, new NoOpErrorHandler());
+
+            // Mocking notification callbacks.
+            var notificationCallbackMock = new Mock<TestNotificationCallbacks>();
+
+            notificationCallbackMock.Setup(nc => nc.TestActivateCallback(It.IsAny<Experiment>(), It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), It.IsAny<Variation>(), It.IsAny<LogEvent>()));
+
+            notificationCallbackMock.Setup(nc => nc.TestAnotherActivateCallback(It.IsAny<Experiment>(), 
+                It.IsAny<string>(), It.IsAny<UserAttributes>(), It.IsAny<Variation>(), It.IsAny<LogEvent>()));
+
+            notificationCallbackMock.Setup(nc => nc.TestFeatureRolloutCallback(It.IsAny<string>(), It.IsAny<string>(),
```

Comment:
please remove feature rollout and feature experiment callbacks.  they will be added later if you can keep them in branch somewhere

Reply:
Removed featurerollout and featureexperiment callbacks. https://github.com/optimizely/csharp-sdk/pull/35/commits/4b096d116d00e04ccdfcee6131f377da6603e3a3